### PR TITLE
Issue #16: Drop PHP 5.6 support

### LINKED ISSUE #16: Drop PHP 5.6 support

### PR DESCRIPTION
**Pull request by tim-bezhashvyly**
Closes #16.

### CODE REVIEW

**Comment by tim-bezhashvyly** [comment]
I had to re-configure build settings at Scrutinizer. Tests are now passing https://scrutinizer-ci.com/g/lizards-and-pumpkins/lib-image-processing-imagick/inspections/cf87c84f-3967-47a0-bc04-3a2f54474fe6

File: tests/Unit/Suites/ImageMagickInscribeStrategyTest.php [hunk 1]
```diff
@@ -28,9 +30,8 @@ public function testImageProcessorStrategyInterfaceIsImplemented()
 
     public function testExceptionIsThrownIfWidthIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image width, got string.');
-        (new ImageMagickInscribeStrategy('foo', 1, 'none'))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new ImageMagickInscribeStrategy('foo', 1, 'none');
```
Comment (Schrank) [inline tests/Unit/Suites/ImageMagickInscribeStrategyTest.php]: PHP is doing this, we can just remove this test, can't we?

File: tests/Unit/Suites/ImageMagickInscribeStrategyTest.php [hunk 2]
```diff
@@ -42,9 +43,8 @@ public function testExceptionIsThrownIfWidthIsNotPositive()
 
     public function testExceptionIsThrownIfHeightIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image height, got string.');
-        (new ImageMagickInscribeStrategy(1, 'foo', 'none'))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new ImageMagickInscribeStrategy(1, 'foo', 'none');
```
Comment (Schrank) [inline tests/Unit/Suites/ImageMagickInscribeStrategyTest.php]: PHP is doing this, we can just remove this test, can't we?

File: tests/Unit/Suites/ImageMagickInscribeStrategyTest.php [hunk 3]
```diff
@@ -54,6 +54,13 @@ public function testExceptionIsThrownIfHeightIsNotPositive()
         (new ImageMagickInscribeStrategy(1, -1, 'none'))->processBinaryImageData('');
     }
 
+
+    public function testExceptionIsThrownIfBackgroundColorIsNotAString()
+    {
+        $this->expectException(\TypeError::class);
+        new ImageMagickInscribeStrategy(1, 1, []);
```
Comment (Schrank) [inline tests/Unit/Suites/ImageMagickInscribeStrategyTest.php]: PHP is doing this, we can just remove this test, can't we?

File: tests/Unit/Suites/ImageMagickResizeStrategyTest.php [hunk 4]
```diff
@@ -27,10 +29,8 @@ public function testImageProcessorStrategyInterfaceIsImplemented()
 
     public function testExceptionIsThrownIfWidthIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image width, got string.');
-
-        (new ImageMagickResizeStrategy('foo', 1))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new ImageMagickResizeStrategy('foo', 1);
```
Comment (Schrank) [inline tests/Unit/Suites/ImageMagickResizeStrategyTest.php]: PHP is doing this, we can just remove this test, can't we?

File: tests/Unit/Suites/ImageMagickResizeStrategyTest.php [hunk 5]
```diff
@@ -43,10 +43,8 @@ public function testExceptionIsThrownIfWidthIsNotPositive()
 
     public function testExceptionIsThrownIfHeightIsNotAnInteger()
     {
-        $this->expectException(InvalidImageDimensionException::class);
-        $this->expectExceptionMessage('Expected integer as image height, got string.');
-
-        (new ImageMagickResizeStrategy(1, 'foo'))->processBinaryImageData('');
+        $this->expectException(\TypeError::class);
+        new ImageMagickResizeStrategy(1, 'foo');
```
Comment (Schrank) [inline tests/Unit/Suites/ImageMagickResizeStrategyTest.php]: PHP is doing this, we can just remove this test, can't we?

File: tests/Unit/Suites/ImageMagickInscribeStrategyTest.php (hunk 1 above)
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/ImageMagickInscribeStrategyTest.php]: We could if signatures would be mandatory.

File: tests/Unit/Suites/ImageMagickInscribeStrategyTest.php (hunk 3 above)
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/ImageMagickInscribeStrategyTest.php]: We could if signatures would be mandatory.

File: tests/Unit/Suites/ImageMagickResizeStrategyTest.php (hunk 4 above)
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/ImageMagickResizeStrategyTest.php]: We could if signatures would be mandatory.

File: tests/Unit/Suites/ImageMagickResizeStrategyTest.php (hunk 5 above)
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/ImageMagickResizeStrategyTest.php]: We could if signatures would be mandatory.

**Pull request merged**